packages/react-sdk/src/db/style.ts

```diff
@@ -0,0 +1,49 @@
+import { z } from "zod";
+import type { StyleProperty, StyleValue } from "@webstudio-is/css-data";
+import { SharedStyleValue } from "@webstudio-is/css-data";
+import { type ComponentName, getComponentMeta } from "../components";
+
+export type PresetStylesItem = {
+  component: ComponentName;
+  property: StyleProperty;
+  value: StyleValue;
+};
+
+// @todo can't figure out how to make component and property to be enum
+export const zPresetStylesItem = z.object({
+  component: z.string(),
+  property: z.string(),
```

Comment:
I'm not sure if you can make this an actual enum. Your `StyleProperty` includes a string index (`[custom: CustomProperty]: { /*...*/}`).

But perhaps smth like this would work:
```ts
z.union([z.enum(properties as [keyof properties]), z.string.regex(/^--/)]);
```

Note that you would have to access those [`properties`](https://github.com/webstudio-is/webstudio-designer/blob/c69c715bb7f16d762bc148702182476b2152e5d5/packages/css-data/src/schema.ts#L2) and plumb them correctly with this call site

Reply:
@kof Why do we have it in types? Shouldn't it be only for css engine?

Reply:
@TrySound not sure what you mean, please explain the full context